### PR TITLE
firejail: Remove unused patches

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Remove unused patches which are already included upstream.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 177646` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/os-specific/linux/firejail/fix-opengl-support.patch [hunk 1]
```diff
@@ -1,7 +0,0 @@
---- a/etc/inc/whitelist-run-common.inc.org	2022-05-07 11:27:32.264849186 +0200
-+++ b/etc/inc/whitelist-run-common.inc	2022-05-07 11:27:55.577778211 +0200
-@@ -13,3 +13,4 @@
- whitelist /run/systemd/resolve/resolv.conf
- whitelist /run/systemd/resolve/stub-resolv.conf
- whitelist /run/udev/data
-+whitelist /run/opengl-driver	# NixOS
```
Comment (klemensn) [inline pkgs/os-specific/linux/firejail/fix-opengl-support.patch:7]: This was added through https://github.com/netblue30/firejail/commit/1c17d794d1c99d1f76ca4140a989b65167edc890
which is NOT contained in the version nixpkgs currently uses
https://github.com/NixOS/nixpkgs/blob/0db6f20cfeb470c1f778442217c39d8202d591bd/pkgs/os-specific/linux/firejail/default.nix#L14

```
$ git branch --contains 1c17d794d1c99d1f76ca4140a989b65167edc890
master
```

File: pkgs/os-specific/linux/firejail/whitelist-nix-profile.patch [hunk 2]
```diff
@@ -1,9 +0,0 @@
---- a/etc/inc/whitelist-common.inc.org	2022-05-06 13:57:17.294206339 +0200
-+++ b/etc/inc/whitelist-common.inc	2022-05-06 13:58:00.108655548 +0200
-@@ -83,3 +83,6 @@
- whitelist ${HOME}/.kde4/share/config/oxygenrc
- whitelist ${HOME}/.kde4/share/icons
- whitelist ${HOME}/.local/share/qt5ct
-+
-+# NixOS specific to resolve binary paths
-+whitelist ${HOME}/.nix-profile
```
Comment (klemensn) [inline pkgs/os-specific/linux/firejail/whitelist-nix-profile.patch:9]: This was added through https://github.com/netblue30/firejail/commit/bd0c18b1bf74bb5b1db2670734c2de8009c5c3d6
which is NOT contained in the version nixpkgs currently uses
https://github.com/NixOS/nixpkgs/blob/0db6f20cfeb470c1f778442217c39d8202d591bd/pkgs/os-specific/linux/firejail/default.nix#L14
```
$ git branch --contains bd0c18b1bf74bb5b1db2670734c2de8009c5c3d6
master
```

**Review by klemensn — CHANGES_REQUESTED** [review]
Reviewing your change would have been so much easier if you simply referenced the upstream changes which nixpkgs supposedly contains already.

https://github.com/netblue30/firejail/commit/1c17d794d1c99d1f76ca4140a989b65167edc890
https://github.com/netblue30/firejail/commit/bd0c18b1bf74bb5b1db2670734c2de8009c5c3d6

**Review by klemensn — APPROVED** [review]

**Comment by klemensn** [comment]
Edit: I made the mistake and checked whether any firejail branch contained the upstream commits (`git branch --contains ...`) making our patches obsolete.
But upstream uses tags not branches so my check made me think we'd still be packaging a version that does not contain those changes, i.e. `git tag --contains ...` showed that the current version `0.9.70` indeed contains both fixes.

That was an error on my side, but still, it could've easily been avoided by providing more details/commentary in your initial PR.

**Pull request merged**